tests/storage/backendstorage.go

```diff
@@ -0,0 +1,131 @@
+
+		By("Expecting the creation of a backend storage PVC with the default storage class")
+		pvc, err := virtClient.CoreV1().PersistentVolumeClaims(vmi.Namespace).Get(context.Background(), backendstorage.PVCForVMI(vmi), metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(pvc.Spec.StorageClassName).NotTo(BeNil())
+		storageClass, err := virtClient.StorageV1().StorageClasses().Get(context.Background(), *pvc.Spec.StorageClassName, metav1.GetOptions{})
+		Expect(storageClass.Annotations).To(HaveKeyWithValue("storageclass.kubernetes.io/is-default-class", "true"))
+	})
+
+	// TODO: create a RequiresRWXFilesystemStorage decorator to remove the skip?
+	It("Should use RWO when RWX is not supported", func() {
+		var storageClass string
+
+		By("Finding a storage class that only supports filesystem in RWO")
+		sps, err := virtClient.CdiClient().CdiV1beta1().StorageProfiles().List(context.Background(), metav1.ListOptions{})
+		Expect(err).NotTo(HaveOccurred()) // Assumes CDI is present. Should we also add a decorator for that?
+		for _, sp := range sps.Items {
+			fsrwo := false
+			fsrwx := false
+			for _, property := range sp.Status.ClaimPropertySets {
+				if property.VolumeMode != nil && *property.VolumeMode == corev1.PersistentVolumeFilesystem {
+					if property.AccessModes != nil {
+						for _, accessMode := range property.AccessModes {
+							if accessMode == corev1.ReadWriteMany {
+								fsrwx = true
+							}
+							if accessMode == corev1.ReadWriteOnce {
+								fsrwo = true
+							}
+						}
+					}
+				}
+			}
+			if fsrwo && !fsrwx && sp.Status.StorageClass != nil {
+				storageClass = *sp.Status.StorageClass
+				break
+			}
+		}
+		if storageClass == "" {
+			Skip("Failed to find a suitable storage class") // See TODO above
+		}
+
+		By("Setting the VM storage class to it")
+		kv := util.GetCurrentKv(virtClient)
+		kv.Spec.Configuration.VMStateStorageClass = storageClass
+		tests.UpdateKubeVirtConfigValueAndWait(kv.Spec.Configuration)
+
+		By("Creating a VMI with persistent TPM")
+		vmi := libvmifact.NewCirros()
+		vmi.Spec.Domain.Devices.TPM = &v1.TPMDevice{Persistent: pointer.P(true)}
+		vmi = tests.RunVMIAndExpectLaunch(vmi, 60)
+
+		By("Expecting the creation of a backend storage PVC with the right storage class")
+		pvc, err := virtClient.CoreV1().PersistentVolumeClaims(vmi.Namespace).Get(context.Background(), backendstorage.PVCForVMI(vmi), metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(pvc.Spec.StorageClassName).NotTo(BeNil())
+		Expect(*pvc.Spec.StorageClassName).To(Equal(storageClass))
+		Expect(pvc.Status.AccessModes).To(HaveLen(1))
+		Expect(pvc.Status.AccessModes[0]).To(Equal(corev1.ReadWriteOnce))
+
+		By("Expecting the VMI to be non-migratable")
+		vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(context.Background(), vmi.Name, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		found := false
+		for _, cond := range vmi.Status.Conditions {
```

Comment:
We can use here the fucntion ([ HasConditionWithStatusAndReason`] (https://github.com/kubevirt/kubevirt/blob/main/pkg/controller/conditions.go#L187)

Reply:
I would have to create a new controller to access that function from test code, and in the end would lose the ability to check the message.
I'd rather keep looping over the conditions, but will do if you and/or others feel strongly about this.